src/libmongoc/tests/test-mongoc-collection.c

```diff
@@ -5630,6 +5630,164 @@ test_create_indexes_with_opts_commitQuorum_post44 (void *unused)
    mongoc_client_destroy (client);
 }
 
+void
+test_insert_reports_id (void)
+{
+   mongoc_client_t *client = test_framework_new_default_client ();
+   mongoc_collection_t *coll = get_test_collection (client, "test_insert_reports_id");
+   bson_error_t error;
+   bool ok;
+
+   // Test inserting one document reports the inserted ID.
+   {
+      bson_t *doc = BCON_NEW ("foo", BCON_UTF8 ("bar"));
```

Comment:
Consider using the helper `tmp_bson` for more concise BSON construction in tests. It is freed by the test runner, so `bson_destroy` is not needed. Example:

```c
bson_t *doc = tmp_bson ("{'foo': 'bar'}");
```